src/Build/Evaluation/Expander.cs

```diff
@@ -1112,14 +1112,34 @@ private static class PropertyExpander<T>
                         return expression;
                     }
 
+                    // Calculate the required size for the stringbuilder
+                    int expectedSize = 0;
+
+                    if(results != null)
+                    {
+                        foreach(object component in results)
+                        {
+                            expectedSize += component.ToString().Length;
```

Comment:
I don't have context on this change, but you are making and discarding temporary strings in a loop. Can that be avoided? Eg., some way to get length without allocating. Or a more crude way to estimate length. Resizing a StringBuilder is not super expensive, since it does not throw out the old block, it just adds another. Since the purpose of StringBuilder is to avoid temp strings, it seems odd to create temp strings in order to prepare a StringBuilder.

Reply:
Ah, I remember this is not a regular StringBuilder. It is supposed to get reused. Have you tried dumping (in some repro cases) what size the returned builder is reaching on Release? Perhaps MaxBuilderSize just needs increasing. That way you get reuse, which is even better than appropriately sizing initially, and was what the code was intended to achieve.